src/database/import.ts

```diff
@@ -190,3 +203,35 @@ export default class DatabaseImporter {
     return [root, key].join("/").replace("//", "/");
   }
 }
+
+/**
+ * Batches chunked JSON data up to the specified byte size limit. The emitted batch is an object
+ * containing chunks keyed by pathname.
+ */
+class BatchChunks extends stream.Transform {
+  private batch: Batch = {};
+  private size = 0;
+
+  constructor(private maxSize: number, opts?: stream.TransformOptions) {
+    super({ ...opts, objectMode: true });
+  }
+
+  _transform(chunk: Data, _: BufferEncoding, callback: stream.TransformCallback) {
+    const chunkSize = JSON.stringify(chunk.json).length;
+    if (this.size + chunkSize > this.maxSize) {
+      this.push(this.batch);
```

Comment:
Can you convert `Batch` into a `Data`?
Find the longest common ancestor of all data.

Reply:
See above.